### PR TITLE
Updating linter for spacing rules

### PR DESCRIPTION
**Pull request by chrisnojima**
In order for this to work we need to move away from using the standalone standard plugin and instead pull its configs into our own and setup rule overrides

@keybase/react-hackers

### CODE REVIEW

File: react-native/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "parser": "babel-eslint",
+    "extends": ["standard", "standard-react"],
+    "rules": {
```
Comment (chrisnojima) [inline react-native/.eslintrc]: we override here. standard made their own plugins to allow this to be loose allowing 0/1 but instead we fallback to the eslint version of one or the other

File: react-native/README.md [hunk 2]
```diff
@@ -94,13 +94,15 @@ See [the objC style guide](../osx/STYLEGUIDE.md)
...
-npm install -g standard
-npm install -g babel-eslint
 npm install -g flow
 ```
 
 See [the JS style guide](standardjs.com)
+We're extending standard to be more strict in some cases (see our .eslintrc)
+
 Visit [standardjs.com](http://standardjs.com/#text-editor-plugins) to find plugins for your editor of choice
+We're including local plugins which requires you to run eslint locally as well (see https://github.com/eslint/eslint/issues/1238)
```
Comment (chrisnojima) [inline react-native/README.md]: Please make sure your editor is doing the right thing. the vim thing is a little tricky

File: react-native/README.md [hunk 3]
```diff
@@ -110,6 +112,15 @@ All files and folders should use the naming convention:
 
 ```
 /this-is-a-folder/a-file.js
+
+/my-component/index.js (common component)
```
Comment (chrisnojima) [inline react-native/README.md]: we should document this new convention

File: react-native/package.json [hunk 4]
```diff
@@ -36,7 +36,12 @@
   },
   "devDependencies": {
     "babel-eslint": "^4.1.5",
-    "redux-devtools": "^2.1.5",
```
Comment (chrisnojima) [inline react-native/package.json]: we never used devtools in mobile

File: react-native/package.json [hunk 5]
```diff
@@ -36,7 +36,12 @@
...
+    "eslint": "^1.9.0",
```
Comment (chrisnojima) [inline react-native/package.json]: now eslint and all these plugins are local

File: react-native/react/actions/config/index.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 'use strict'
 
 import * as Constants from '../../constants/config'
-import { autoLogin } from '../login'
```
Comment (chrisnojima) [inline react-native/react/actions/config/index.js]: the rest is fixes that came out of this linting

File: react-native/react/actions/login.js [hunk 7]
```diff
@@ -98,8 +98,8 @@ function qrGenerate (code) {
   qr.addData(code)
   qr.make()
   let tag = qr.createImgTag(10)
-  const [ , src, , ] = tag.split(' ')
```
Comment (chrisnojima) [inline react-native/react/actions/login.js]: The parsers really don't like this in general so i'm taking it out

File: react-native/README.md (hunk 2 above)
Comment (MarcoPolo) [inline react-native/README.md]: I use vim, what should I do?

Comment (chrisnojima) [inline react-native/README.md]: I added a comment about con already below that?

Comment (MarcoPolo) [inline react-native/README.md]: ha, just saw the line below nvm

**Comment by MarcoPolo** [comment]
:+1:

**Comment by chrisnojima** [comment]
@cbj it's also technically possible to npm install -g those plugins and leave your editor using the globals, that'll work too

**Pull request merged**